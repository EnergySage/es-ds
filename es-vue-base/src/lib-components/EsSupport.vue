<template>
    <div
        class="d-flex support-holder align-items-center"
        v-bind="$attrs"
        v-on="$listeners">
        <div class="image-holder mr-100">
            <a
                target="_blank"
                aria-label="Help"
                :href="link">
                <b-img
                    width="64px"
                    height="64px"
                    :class="`image image-bg-${variant}`"
                    :src="src"
                    alt="Help Image" />
            </a>
        </div>
        <div class="text-holder d-flex flex-column">
            <div class="title">
                <div class="h5 mb-25 font-size-base font-weight-bold">
                    <slot
                        v-if="hasTitle"
                        name="title" />
                    <template v-else>
                        Need help signing up?
                    </template>
                </div>
            </div>
            <div class="link">
                <a
                    target="_blank"
                    class="font-size-75 supportLink"
                    :href="link">
                    <slot
                        v-if="hasLinkCopy"
                        name="linkCopy" />
                    <template v-else>
                        Schedule a free call with your EnergySage Advisor.
                    </template>
                </a>
            </div>
        </div>
    </div>
</template>

<script lang="js">
import { BImg } from 'bootstrap-vue';

export default {
    name: 'EsSupport',
    components: {
        BImg,
    },
    props: {
        /**
         * Link
         */
        link: {
            default: '',
            type: String,
            required: true,
        },
        /**
         * Image src
         *
         */
        src: {
            type: String,
            required: true,
        },
        /**
         * Support Variant
         */
        variant: {
            type: String,
            required: false,
            default: 'warm',
            validator: (val) => ['warm', 'cool'].includes(val),
        },
    },
    computed: {
        hasTitle() {
            return !!this.$slots.title;
        },
        hasLinkCopy() {
            return !!this.$slots.linkCopy;
        },
    },
};
</script>

<style lang="scss" scoped>
@use "~@energysage/es-bs-base/scss/variables" as variables;

.support-holder {
    max-width: 325px;

    .image-holder {
        height: 64px;
        width: 64px;

        .image {
            border-radius: 2rem;
            object-fit: contain;

            &.image-bg-warm {
                background: linear-gradient(141.22deg, variables.$yellow 8.76%, variables.$pink 100%);
            }

            &.image-bg-cool {
                background: variables.$teal-300;
            }
        }
    }
}
</style>
