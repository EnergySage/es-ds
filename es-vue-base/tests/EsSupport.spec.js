import { mount } from '@vue/test-utils';
import EsSupport from '@/src/lib-components/EsSupport.vue';
import jestVue from '@/tests/jest.vue.config';
import { BImg } from 'bootstrap-vue';

describe('EsSupport', () => {
    // Basic test; should exist for most components
    const slots = {
        default: 'Need help signing up?',
    };

    test('<EsSupport />', async () => {
        const wrapper = mount(EsSupport, {
            ...jestVue,
            slots,
            propsData: {
                link: 'https://www.google.com/',
                src: 'https://via.placeholder.com/200x100',
            },
        });
        const a11y = await axe(wrapper.element);

        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
        expect(a11y).toHaveNoViolations();
    });

    // Test Support Title works
    test('Support title slot exists', async () => {
        const wrapper = mount(EsSupport, {
            ...jestVue,
            slots,
            propsData: {
                link: 'https://www.google.com/',
                src: 'https://via.placeholder.com/200x100',
            },
        });

        const titleSlot = wrapper.vm.$slots.default;
        expect(titleSlot[0].text).toBe('Need help signing up?');
        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
    });

    // Support text exists
    test('Support text exists', async () => {
        const wrapper = mount(EsSupport, {
            ...jestVue,
            slots,
            propsData: {
                link: 'https://www.google.com/',
                src: 'https://via.placeholder.com/200x100',
            },
        });

        const supportLink = wrapper.find('.supportLink');
        expect(supportLink.text()).toBe('Schedule a free call with your EnergySage Advisor.');
        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
    });

    // Image component exists
    test('Image exists', async () => {
        const wrapper = mount(EsSupport, {
            ...jestVue,
            slots,
            propsData: {
                link: 'https://www.google.com/',
                src: 'https://via.placeholder.com/200x100',
            },
        });

        expect(wrapper.findComponent(BImg).exists()).toBe(true);
        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
    });
});
