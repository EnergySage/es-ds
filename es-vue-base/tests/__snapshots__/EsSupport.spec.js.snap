// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EsSupport <EsSupport /> 1`] = `
"<div class="d-flex support-holder align-items-center">
  <div class="image-holder mr-100"><a target="_blank" aria-label="Help" href="https://www.google.com/"><img src="https://via.placeholder.com/200x100" alt="Help Image" width="64" height="64" class="image image-bg-warm"></a></div>
  <div class="text-holder d-flex flex-column">
    <div class="title">
      <div class="h5 mb-25 font-size-base font-weight-bold">
        Need help signing up?
      </div>
    </div>
    <div class="link"><a target="_blank" href="https://www.google.com/" class="supportLink">
        Schedule a free call with your EnergySage Advisor.
      </a></div>
  </div>
</div>"
`;

exports[`EsSupport <EsSupport variant="cool" /> 1`] = `
"<div class="d-flex support-holder align-items-center">
  <div class="image-holder mr-100"><a target="_blank" aria-label="Help" href="https://www.google.com/"><img src="https://via.placeholder.com/200x100" alt="Help Image" width="64" height="64" class="image image-bg-cool"></a></div>
  <div class="text-holder d-flex flex-column">
    <div class="title">
      <div class="h5 mb-25 font-size-base font-weight-bold">
        Need help signing up?
      </div>
    </div>
    <div class="link"><a target="_blank" href="https://www.google.com/" class="supportLink">
        Schedule a free call with your EnergySage Advisor.
      </a></div>
  </div>
</div>"
`;

exports[`EsSupport Image exists 1`] = `
"<div class="d-flex support-holder align-items-center">
  <div class="image-holder mr-100"><a target="_blank" aria-label="Help" href="https://www.google.com/"><img src="https://via.placeholder.com/200x100" alt="Help Image" width="64" height="64" class="image image-bg-warm"></a></div>
  <div class="text-holder d-flex flex-column">
    <div class="title">
      <div class="h5 mb-25 font-size-base font-weight-bold">
        Need help signing up?
      </div>
    </div>
    <div class="link"><a target="_blank" href="https://www.google.com/" class="supportLink">
        Schedule a free call with your EnergySage Advisor.
      </a></div>
  </div>
</div>"
`;

exports[`EsSupport Support text exists 1`] = `
"<div class="d-flex support-holder align-items-center">
  <div class="image-holder mr-100"><a target="_blank" aria-label="Help" href="https://www.google.com/"><img src="https://via.placeholder.com/200x100" alt="Help Image" width="64" height="64" class="image image-bg-warm"></a></div>
  <div class="text-holder d-flex flex-column">
    <div class="title">
      <div class="h5 mb-25 font-size-base font-weight-bold">
        Need help signing up?
      </div>
    </div>
    <div class="link"><a target="_blank" href="https://www.google.com/" class="supportLink">
        Schedule a free call with your EnergySage Advisor.
      </a></div>
  </div>
</div>"
`;

exports[`EsSupport Support title slot exists 1`] = `
"<div class="d-flex support-holder align-items-center">
  <div class="image-holder mr-100"><a target="_blank" aria-label="Help" href="https://www.google.com/"><img src="https://via.placeholder.com/200x100" alt="Help Image" width="64" height="64" class="image image-bg-warm"></a></div>
  <div class="text-holder d-flex flex-column">
    <div class="title">
      <div class="h5 mb-25 font-size-base font-weight-bold">
        Need help signing up?
      </div>
    </div>
    <div class="link"><a target="_blank" href="https://www.google.com/" class="supportLink">
        Schedule a free call with your EnergySage Advisor.
      </a></div>
  </div>
</div>"
`;
